client/components/layout/UserForm.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button, Container, TextField } from '@mui/material';
+import { Box } from '@mui/system';
+import axios from 'axios';
+import { useState } from 'react';
+
+import { useAuthDispatch } from 'client/contexts/auth';
+import { UserProfile } from 'server/types/User';
+
+interface Props {
+  user: UserProfile;
+  setUser: React.Dispatch<React.SetStateAction<UserProfile>>;
+}
+
+const UserForm:React.FC<Props> = ({ user, setUser }) => {
+  const authDispatch = useAuthDispatch();
+
+  const [userData, setUserData] = useState({
+    bio: user.bio || '',
+    twitter_username: user.twitter_username || '',
+    linkedin_url: user.linkedin_url || '',
+    github_username: user.github_username || '',
+    website: user.website || '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isFormVisible, setIsFormVisible] = useState(true);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setUserData(prevState => ({ ...prevState, [name]: value }));
+  };
+
+  const handleSubmit:React.FormEventHandler<HTMLFormElement> = async event => {
+    event.preventDefault();
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+    try {
+      const userId = user.id;
+
+      await axios.patch(`/api/users/${userId}`, userData);
+      authDispatch({ type: 'SET_AUTHED_USER', user: { ...user, ...userData }});
+      setUser({
+        ...user,
+        ...userData,
+      });
+      setIsFormVisible(false);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  if (!isFormVisible) {
+    return null;
+  }
+
+  return (
+    <Container maxWidth="sm">
+      <form onSubmit={handleSubmit}>
+        <Box display="flex" flexDirection="column" gap={2}>
+          <TextField
+            id="outlined-multiline-static"
```

Comment:
what's the purpose of this id attribute?

Reply:
Looks like nothing